### DISCUSSION TITLE
Descriptions of Joomla Fields

### DISCUSSION THREAD

**Discussion by leoalv** [joomla/joomla-cms#32813] [General] [answered]
Until Joomla 3.x we have had the descriptions of the fields, which in the interface were shown in tooltips. In my opinion they were of great help so that users and integrators know the function of each field. In Joomla 4 these descriptions no longer exist, they were removed from the core extensions.

Given this my questions are: why did they remove it? will it be replaced by another? Or do we just say goodbye?

PS: If you add descriptions in the Joomla 4 fields they show up, but they are a complete disaster when you have great explanations.

**Comment by chmst** [comment, marked as answer]
There were two reasons for removin tooltips: 
They showed up on hover, so not on touchscreens and not for keyboard users which affects accessibility.
And many tooltips repeated only what's obvious. 

Important information should be displayed below the respective field in J4.

It is a challenge to write informative and short explanations where it is required, that is true.

**Comment by krileon** [comment]
Tooltips should only exist for non-essential information and only if it's actual relevant information. A lot of tooltips just parroted the obvious. For example in Article Create the Title input literally has a tooltip of "Title". The Status input has a tooltip of "Set publication status". Both of those and 99% of the other tooltips throughout backend were just pointless.

For locations with actual useful Tooltips they were being used for displaying essential information for completing a form. Essential information should just always be visible. Example below.

https://getbootstrap.com/docs/5.0/forms/overview/#form-text

Notice the descriptive text for password is directly under the input as it's necessary information for completing the field. The end result is most tooltips shouldn't have existed and those that were necessary should be moved under the input. Any other non-essential information could then just be an actual tooltip that needs to function on hover, on click, and be tab indexed so they're accessible to mouse, mobile, and keyboard navigations.

**Comment by leoalv** [comment]
My question is because I am developing some extensions for Joomla 4; and migrating others from J3. For now the way is to make an extreme summary of the necessary descriptions and say goodbye to the redundant ones (in fact these I don't use them).

If there is any other plan or project for the descriptions speak now or shut up forever.

**Reply by brianteeman** [reply]
If you add a description to your field it will be displayed below the field itself.
